Sources/Build/Triple.swift

```diff
@@ -22,61 +22,155 @@ public struct Triple {
     public let arch: Arch
     public let vendor: Vendor
     public let os: OS
+    public let osVersion: Version?
     public let abi: ABI
 
     public enum Error: Swift.Error {
         case badFormat
-        case unknownArch
         case unknownOS
     }
 
     public enum Arch: String {
+        case unknown
+
+        case arm
+        case arm64
+        case i386
         case x86_64
-        case ppc64le
+        case powerpc64
+        case powerpc64le
         case s390x
-        case aarch64
+
+        // TODO: Split out into a separate SubArch enum?
         case armv7
-        case arm
+        case armv7k
+        case armv7s
     }
 
     public enum Vendor: String {
         case unknown
-        case apple
+
+        /// Indicates Apple Inc., the vendor of iOS, macOS, tvOS, and watchOS.
+        case Apple = "apple"
+
+        /// Indicates vendor-agnostic platform "PC"-type platform.
+        case PC = "pc"
+
+        /// Indicates Sony Computer Entertainment, Inc., the vendor of PS4 OS.
+        case SCEI = "scei"
```

Comment:
I would keep the previous casing, I prefer downcasing initial acronyms (although I think Swift hasn't formally decided on a convention).

Reply:
> NOTE: There are some subtle differences between Swift's expected -target triples and LLVM's triples.

For example?

Reply:
```
$ swiftc -c -### t.swift
.... -target x86_64-apple-darwin17.6.0 ...

$ clang -c -### t.c
... "-triple" "x86_64-apple-macosx10.13.0" ...
```

Reply:
They're interchangeable enough. Swift canonicalizes pretty aggressively internally when it cares.

Reply:
Note that "darwin" in an LLVM triple, whether Clang or Swift, is assumed to mean "macosx", not "any Apple platform".